Bottom: 43fbcc5410fdb6dea4a55a49b86a3463b4db247d
Top:    f8f468f2d480812a7b628603961f0cb10eb3e9eb
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2014-07-20 12:57:32 -0400

Refresh of tyler

---

diff --git a/arch/arm/boot/dts/sun7i-a20-cubietruck.dts b/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
index dd03c64..df6414e 100644
--- a/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
+++ b/arch/arm/boot/dts/sun7i-a20-cubietruck.dts
@@ -120,7 +120,7 @@
 
 		iis0: iis@01c22000 {
 			pinctrl-names = "default";
-			/*pinctrl-0 = <&i2s0_pins_a>;*/
+			/*pinctrl-0 = <&iis0_pins_a>;*/
 			status = "okay";
 		};
 
@@ -199,7 +199,7 @@
 		simple-audio-card,mclk-fs = <256>;
 
 		simple-audio-card,cpu {
-			sound-dai = <&i2s0>;
+			sound-dai = <&iis0>;
 		};
 
 		simple-audio-card,codec {
diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index 61928aa..db6d51b 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
+++ b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -875,10 +875,11 @@
 		};
 
 		iis0: iis@01c22000 {
+			#sound-dai-cells = <0>;
 			compatible = "allwinner,sun7i-a20-iis";
 			reg = <0x01C22000 0x40>;
 			interrupts = <0 16 4>;
-			clocks = <&pll2>, <&apb0_gates 3>, <&iis0_clk>;
+			clocks = <&pll2 0>, <&apb0_gates 3>, <&iis0_clk>;
 			clock-names = "pll2", "apb", "iis";
 			dmas = <&dma 0 3>, <&dma 0 3>;
 			dma-names = "rx", "tx";
@@ -886,10 +887,11 @@
 		};
 
 		iis1: iis@01c22400 {
+			#sound-dai-cells = <0>;
 			compatible = "allwinner,sun7i-a20-iis";
 			reg = <0x01C22400 0x40>;
 			interrupts = <0 87 4>;
-			clocks = <&pll2>, <&apb0_gates 4>, <&iis1_clk>;
+			clocks = <&pll2 0>, <&apb0_gates 4>, <&iis1_clk>;
 			clock-names = "pll2", "apb", "iis";
 			dmas = <&dma 0 4>, <&dma 0 4>;
 			dma-names = "rx", "tx";
@@ -897,10 +899,11 @@
 		};
 
 		iis2: iis@01c24400 {
+			#sound-dai-cells = <0>;
 			compatible = "allwinner,sun7i-a20-iis";
 			reg = <0x01C24400 0x40>;
 			interrupts = <0 90 4>;
-			clocks = <&pll2>, <&apb0_gates 8>, <&iis2_clk>;
+			clocks = <&pll2 0>, <&apb0_gates 8>, <&iis2_clk>;
 			clock-names = "pll2", "apb", "iis";
 			dmas = <&dma 0 6>, <&dma 0 6>;
 			dma-names = "rx", "tx";
diff --git a/sound/soc/sunxi/sunxi-i2s.c b/sound/soc/sunxi/sunxi-i2s.c
index 3b3873a..edba590 100644
--- a/sound/soc/sunxi/sunxi-i2s.c
+++ b/sound/soc/sunxi/sunxi-i2s.c
@@ -530,10 +530,10 @@ static const struct snd_soc_dapm_widget i2s_dapm_widgets[] = {
 };
 
 static const struct of_device_id sunxi_i2s_of_match[] = {
-	{ .compatible = "allwinner,sun4i-a10a-i2s", .data = (void *)SUN4A},
-	{ .compatible = "allwinner,sun4i-a10-i2s", .data = (void *)SUN4I},
-	{ .compatible = "allwinner,sun5i-a13-i2s", .data = (void *)SUN5I},
-	{ .compatible = "allwinner,sun7i-a20-i2s", .data = (void *)SUN7I},
+	{ .compatible = "allwinner,sun4i-a10a-iis", .data = (void *)SUN4A},
+	{ .compatible = "allwinner,sun4i-a10-iis", .data = (void *)SUN4I},
+	{ .compatible = "allwinner,sun5i-a13-iis", .data = (void *)SUN5I},
+	{ .compatible = "allwinner,sun7i-a20-iis", .data = (void *)SUN7I},
 	{}
 };
 MODULE_DEVICE_TABLE(of, sunxi_i2s_of_match);
@@ -588,9 +588,9 @@ static int sunxi_i2s_probe(struct platform_device *pdev)
 		dev_err(dev, "failed to get pll2 clock\n");
 		return PTR_ERR(priv->clk_pll2);
 	}
-	priv->clk_module = devm_clk_get(dev, "i2s");
+	priv->clk_module = devm_clk_get(dev, "iis");
 	if (IS_ERR(priv->clk_module)) {
-		dev_err(dev, "failed to get i2s clock\n");
+		dev_err(dev, "failed to get iis clock\n");
 		return PTR_ERR(priv->clk_module);
 	}
